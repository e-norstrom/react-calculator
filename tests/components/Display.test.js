"use strict";

var _react = _interopRequireDefault(require("react"));

var _Display = _interopRequireDefault(require("../../components/Display"));

var _enzyme = require("enzyme");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

test('It should render the empty Display correctly', function () {
  var wrapper = (0, _enzyme.shallow)( /*#__PURE__*/_react["default"].createElement(_Display["default"], null));
  expect(wrapper).toMatchSnapshot();
});
test('It should render input correctly', function () {
  var input = ['1', '3', '+', '0'];
  var wrapper = (0, _enzyme.shallow)( /*#__PURE__*/_react["default"].createElement(_Display["default"], {
    input: input
  }));
  expect(wrapper).toMatchSnapshot();
});
test('It should render result correctly', function () {
  var result = ['53424.00912'];
  var wrapper = (0, _enzyme.shallow)( /*#__PURE__*/_react["default"].createElement(_Display["default"], {
    result: result
  }));
  expect(wrapper).toMatchSnapshot();
});