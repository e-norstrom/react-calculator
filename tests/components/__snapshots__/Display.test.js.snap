// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render input correctly 1`] = `
<div
  className="display"
>
  <div
    className="display-result"
    type="numbers"
  >
     
  </div>
  <div
    className="display-numbers"
    id="display"
    type="numbers"
  >
    1
    3
    +
    0
  </div>
</div>
`;

exports[`It should render result correctly 1`] = `
<div
  className="display"
>
  <div
    className="display-result"
    type="numbers"
  >
     
    53424.00912
  </div>
  <div
    className="display-numbers"
    id="display"
    type="numbers"
  />
</div>
`;

exports[`It should render the empty Display correctly 1`] = `
<div
  className="display"
>
  <div
    className="display-result"
    type="numbers"
  >
     
  </div>
  <div
    className="display-numbers"
    id="display"
    type="numbers"
  />
</div>
`;
